app/models/binary/DataSetService.scala

```diff
@@ -82,6 +83,20 @@ object DataSetService extends FoxImplicits with LazyLogging {
     }
   }
 
+  def deactivateDataSources(dataStoreName: String)(implicit ctx: DBAccessContext) = {
+    DataSetDAO.update(
+      Json.obj("dataStoreInfo.name" -> dataStoreName),
+      Json.obj(
+        "$set" -> Json.obj(
+          "isActive" -> false,
+          "dataSource.status" -> "No longer available on datastore."),
+        "$unset" -> Json.obj(
```

Comment:
is the `$unset` needed?

Reply:
I think so, since when we deserialize [here](https://github.com/scalableminds/braingames-libs/blob/master/braingames-binary/src/main/scala/com/scalableminds/braingames/binary/models/datasource/InboxDataSource.scala#L22) we would succeed, although we should not.

Reply:
makes sense :+1: please add a comment though